### PR TITLE
release to edge on push to main

### PR DESCRIPTION
**Pull request by simskij**
## Changes

- Splits the CI into two different workflows, one for PRs and one for releases.
- Replaces the legacy library check with the one provided by `charming-actions`.
- Introduces charm release automation using the RC version of the `charming-actions`.
- Makes all git checkouts shallow to save some time while building.
- Cleans up some old cruft related to `matrix` builds that we don't use anymore.
- Harmonizes the job names to match our other repositories.

## Test Instructions

- Verify that the workflow passes for the PR.

No further testing possible as the release automation only is executed on pushes to main.

### CODE REVIEW

**Review by balbirthomas — APPROVED** [review]

File: .github/workflows/pull-request.yaml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      - name: Set up Python 3.5
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.5
+      - name: Install dependencies
+        run: python3 -m pip install tox
+      - name: Run static analysis for /lib for 3.5
+        run: tox -vve static-lib
+
+  static-charm:
```
[the remark this answers is not in the document]
Comment (simskij) [reply .github/workflows/pull-request.yaml:40]: i disagree. static-libs is also a static analysis. all envs run under static-charm in one way or another have with the charm to do (either it's the charm code itself, or its tests).

[the remark this answers is not in the document]
Comment (sed-i) [reply .github/workflows/pull-request.yaml:40]: Not sure what you mean. This section is clearly names after one of the underlying tox envs: `static-charm`.
However, this section is about static analysis for more than the charm:
```
        run: tox -vve static-charm
        run: tox -vve static-unit
        run: tox -vve static-integration
```

[the remark this answers is not in the document]
Comment (sed-i) [reply .github/workflows/pull-request.yaml:40]: Oh ok got it. Sure.

**Review by sed-i — APPROVED** [review]

File: .github/workflows/pull-request.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (simskij) [reply .github/workflows/pull-request.yaml:40]: Bad phrasing on my behalf. Sorry about that.

**Pull request merged**